### PR TITLE
Enabling black for Code Style Check in AutoMM

### PR DESCRIPTION
**Pull request by zhiqiangdon**
As contributors increase, we need linters to ensure the code format consistency in AutoMM. Automatic linting is necessary to guarantee code quality and save reviewers' time from checking the code style. In this PR, we turn on linter black (https://github.com/psf/black)

### CODE REVIEW

File: text/setup.py [hunk 1]
```diff
@@ -43,6 +43,7 @@
     'autogluon-contrib-nlp==0.0.1b20220208',
     'nlpaug>=1.1.10,<2.0.0',
     'nltk>=3.4.5,<4.0.0'
+    'black~=22.0,>=22.3',
```
Comment (sxjscience) [inline text/setup.py:46]: Let's not add it to text/setup.py. We may add it to Jenkins directly. Users of autogluon.text / AutoMM do not need to install black.

File: Jenkinsfile [hunk 2]
```diff
@@ -190,6 +190,8 @@ stage("Unit Test") {
           # launch different process for each test to make sure memory is released
           python3 -m pip install --upgrade pytest-xdist
 
+          check_dirs := text/src/autogluon/text/automm tests/unittests/automm
```
Comment (sxjscience) [inline Jenkinsfile:193]: Use

```python
export AUTOMM_LINT_DIRS=text/src/autogluon/text/automm tests/unittests/automm
black --check --preview $(AUTOMM_LINT_DIRS)
```

File: Jenkinsfile [hunk 3]
```diff
@@ -190,6 +190,9 @@ stage("Unit Test") {
           # launch different process for each test to make sure memory is released
           python3 -m pip install --upgrade pytest-xdist
 
+          python3 -m pip install 'black~=22.0,>=22.3'
+          export AUTOMM_LINT_DIRS=text/src/autogluon/text/automm tests/unittests/automm
+          black --check --preview $(AUTOMM_LINT_DIRS)
```
Comment (sxjscience) [inline Jenkinsfile:195]: Use ```black --check --preview ${AUTOMM_LINT_DIRS}```

File: text/setup.py (hunk 1 above)
Comment (yinweisu) [reply text/setup.py]: Another way around it is add the dependency to text[test] for example. And we can do `pip install text[test]` in Jenkins.

File: Jenkinsfile [hunk 4]
```diff
@@ -190,6 +190,8 @@ stage("Unit Test") {
           # launch different process for each test to make sure memory is released
           python3 -m pip install --upgrade pytest-xdist
 
+          python3 -m pip install 'black~=22.0,>=22.3'
+          black --check --preview text/src/autogluon/text/automm
```
Comment (sxjscience) [inline Jenkinsfile:194]: Change to 
```
black --check --diff text/src/autogluon/text/automm
```

**Review by sxjscience — APPROVED** [review]
I've skimmed through the changes and it looks good to me. Thanks for adding lint support!!!

**Review by bryanyzhu — APPROVED** [review]
Awesome PR, this is great!

**Review by yinweisu — APPROVED** [review]
Nice work!! I'll run a local black style format and merge in this code after it's been merged into master. Remember to notify anyone currently working on AutoMM PRs to do the same ;)

**Comment by szha** [comment]
Job PR-1783-10 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1783/10/index.html

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+[tool.black]
+line-length = 119
```
Comment (Innixma) [inline pyproject.toml:2]: Are we sure we want 119? Thoughts on 139 or 159? I personally use 159 in my IDE.

File: Jenkinsfile [hunk 6]
```diff
@@ -190,6 +190,8 @@ stage("Unit Test") {
           # launch different process for each test to make sure memory is released
           python3 -m pip install --upgrade pytest-xdist
 
+          python3 -m pip install 'black~=22.0,>=22.3'
```
Comment (Innixma) [inline Jenkinsfile:193]: Isn't `~=22.0` redundant?

File: Jenkinsfile [hunk 7]
```diff
@@ -190,6 +190,8 @@ stage("Unit Test") {
...
+          black --check --preview --diff text/src/autogluon/text/automm
```
Comment (Innixma) [inline Jenkinsfile:194]: Will this fail CI if it doesn't align with black formatting? Should we add a README on this? Unclear currently steps someone needs to take to satisfy formatting in their PR.

File: pyproject.toml [hunk 8]
```diff
@@ -0,0 +1,3 @@
...
+target-version = ['py38']
```
Comment (Innixma) [inline pyproject.toml:3]: Should we instead have:

`target-version = ['py37', 'py38', 'py39']`

As is done in sklearn: https://github.com/scikit-learn/scikit-learn/blob/main/pyproject.toml

File: pyproject.toml (hunk 5 above)
Comment (sxjscience) [reply pyproject.toml:2]: 120 is the default line length for Intellij. 139/159 will be too long for Python. Actually PEP8 will ask for 79: https://legacy.python.org/dev/peps/pep-0008/#maximum-line-length .  It is reasonable to use 119 for AutoMM.

Comment (zhiqiangdon) [reply pyproject.toml:2]: Huggingface uses 119 (https://github.com/huggingface/transformers/blob/main/pyproject.toml#L2)
sklearn uses 88 (https://github.com/scikit-learn/scikit-learn/blob/main/pyproject.toml#L18)
PEP 8 suggests 79. The default line length in Pycharm is 120. Considering all these, I think we'd better not use too long lines.

File: pyproject.toml (hunk 8 above)
Comment (zhiqiangdon) [reply pyproject.toml]: Thanks. Done.

File: pyproject.toml (hunk 5 above)
Comment (Innixma) [reply pyproject.toml:2]: Ok, we can try 119 or 120 and see.

File: Jenkinsfile (hunk 6 above)
Comment (zhiqiangdon) [reply Jenkinsfile]: I referred to here: https://github.com/huggingface/transformers/blob/main/setup.py#L101
Not sure whether they have some special considerations.

**Comment by szha** [comment]
Job PR-1783-11 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1783/11/index.html

**Comment by szha** [comment]
Job PR-1783-12 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1783/12/index.html

File: Jenkinsfile (hunk 7 above)
Comment (sxjscience) [reply Jenkinsfile:193]: Let’s add a README in the next PR and merge this in first.

**Pull request merged**